src/util/NuulsUploader.hpp

```diff
@@ -0,0 +1,22 @@
+#include "common/Channel.hpp"
+#include "widgets/helper/ResizingTextEdit.hpp"
+
+#include <QMimeData>
+#include <QString>
+
+namespace chatterino {
+struct TypedBytes {
+    QByteArray data;
+    QString type;
+};
+void upload(QByteArray imageData, ChannelPtr channel,
+            ResizingTextEdit &textEdit, std::string format);
+void upload(TypedBytes imageData, ChannelPtr channel,
+            ResizingTextEdit &textEdit);
+void upload(const QMimeData *source, ChannelPtr channel,
+            ResizingTextEdit &outputTextEdit);
+}  // namespace chatterino
+
+namespace {
+QString getImageFileFormat(QString path);
+}
```

Comment:
this should not be in the header file, if it's an exported function just put it in the chatterino namespace. but from my investigations, it's only used in the `NuulsUploader.cpp` file